e2e/internal/e2e/imageverify.go

```diff
@@ -0,0 +1,406 @@
+		}
+	}
+
+	// Verify any apps
+	for _, app := range dfd.Apps {
+		// %apphelp
+		if app.Help != nil {
+			helpPath := filepath.Join(imagePath, `/scif/apps/`, app.Name, `/scif/runscript.help`)
+			if !fileExists(t, helpPath) {
+				t.Fatalf("unexpected failure in app %v: Script %v does not exist in app", app.Name, helpPath)
+			}
+
+			if err := verifyHelp(t, helpPath, app.Help); err != nil {
+				t.Fatalf("unexpected failure in app %v: app help message: %v", app.Name, err)
+			}
+		}
+
+		// %appenv
+		if app.Env != nil {
+			if err := verifyEnv(t, cmdPath, imagePath, app.Env, []string{"--app", app.Name}); err != nil {
+				t.Fatalf("unexpected failure in app %v: Env in app is incorrect: %v", app.Name, err)
+			}
+		}
+
+		// %applabels
+		if app.Labels != nil {
+			if err := verifyAppLabels(t, imagePath, app.Name, app.Labels); err != nil {
+				t.Fatalf("unexpected failure in app %v: Labels in app are incorrect: %v", app.Name, err)
+			}
+		}
+
+		// %appfiles
+		for _, p := range app.Files {
+			var file string
+			if p.Src == "" {
+				file = p.Src
+			} else {
+				file = p.Dst
+			}
+
+			if !fileExists(t, filepath.Join(imagePath, "/scif/apps/", app.Name, file)) {
+				t.Fatalf("unexpected failure in app %v: File %v does not exist in app", app.Name, file)
+			}
+
+			if err := verifyFile(t, p.Src, filepath.Join(imagePath, "/scif/apps/", app.Name, file)); err != nil {
+				t.Fatalf("unexpected failure in app %v: File %v: %v", app.Name, file, err)
+			}
+		}
+
+		// %appInstall
+		for _, file := range app.Install {
+			if !fileExists(t, filepath.Join(imagePath, "/scif/apps/", app.Name, file)) {
+				t.Fatalf("unexpected failure in app %v: %%Install generated file %v does not exist in container", app.Name, file)
+			}
+		}
+
+		// %appRun
+		if app.Run != nil {
+			scriptPath := filepath.Join(imagePath, "/scif/apps/", app.Name, "scif/runscript")
+			if !fileExists(t, scriptPath) {
+				t.Fatalf("unexpected failure in app %v: Script %v does not exist in app", app.Name, scriptPath)
+			}
+
+			if err := verifyScript(t, scriptPath, app.Run); err != nil {
+				t.Fatalf("unexpected failure in app %v: runscript: %v", app.Name, err)
+			}
+		}
+
+		// %appTest
+		if app.Test != nil {
+			scriptPath := filepath.Join(imagePath, "/scif/apps/", app.Name, "scif/test")
+			if !fileExists(t, scriptPath) {
+				t.Fatalf("unexpected failure in app %v: Script %v does not exist in app", app.Name, scriptPath)
+			}
+
+			if err := verifyScript(t, scriptPath, app.Test); err != nil {
+				t.Fatalf("unexpected failure in app %v: test script: %v", app.Name, err)
+			}
+		}
+	}
+
+}
+
+func fileExists(t *testing.T, path string) bool {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false
+	} else if err != nil {
+		t.Fatalf("While stating file: %v", err)
+	}
+
+	return true
+}
+
+func verifyFile(t *testing.T, original, copy string) error {
+	ofi, err := os.Stat(original)
```

Comment:
https://godoc.org/os#SameFile

This seems problematic for big files but I'm not sure we need to optimize this yet.

Reply:
Seems like `SameFile()` is essentially checking if the files are stored in the same inodes and not checking the content? I think we want to check the content in our case. This is a potential alternative https://godoc.org/github.com/udhos/equalfile

Reply:
Yeah, I just noticed that you have part of your function doing what os.SameFile does as well and thought you could substitute that in.